integration_tests/__tests__/json_reporter-test.js

```diff
@@ -9,10 +9,39 @@
 
 const runJest = require('../runJest');
 const skipOnWindows = require('skipOnWindows');
+const fs = require('fs');
+const path = require('path');
 
 describe('JSON Reporter', () => {
   skipOnWindows.suite();
 
+  it('writes test result to sum.result.json', () => {
+    const outputFileName = 'sum.result.json';
+    const outputFilePath = path.join(
+      process.cwd(),
+      'integration_tests/json_reporter/',
+      outputFileName
+    );
+    runJest('json_reporter', ['--json', `--jsonOutputFile=${outputFileName}`]);
+    const testOutput = fs.readFileSync(outputFilePath, 'UTF-8', (err, res) => {
```

Comment:
you are reading the file synchronously, the third argument is unused.

Also please use `utf8` for consistency reasons with the rest of Jest.